File: android/src/main/java/com/margelo/nitro/fs2/utils/StreamError.kt
package com.margelo.nitro.fs2.utils

sealed class StreamError : Exception() {
    data class NotFound(val path: String) : StreamError() {
        override val message: String = "ENOENT: File does not exist: $path"
    }
    
    data class AccessDenied(val path: String) : StreamError() {
        override val message: String = "EACCES: Permission denied: $path"
    }
    
    data class IOError(override val message: String) : StreamError()
    
    data class StorageError(val reason: String) : StreamError() {
        override val message: String = "Storage error: $reason"
    }
    
    data class StreamClosed(val streamId: String) : StreamError() {
        override val message: String = "EPIPE: Stream is closed: $streamId"
    }
    
    data class StreamInactive(val streamId: String) : StreamError() {
        override val message: String = "EPIPE: Stream is not active: $streamId"
    }
    
    data class InvalidStream(val streamId: String) : StreamError() {
        override val message: String = "ENOENT: No such stream: $streamId"
    }
    
    data class BufferError(override val message: String) : StreamError()
} 

File: ios/StreamError.swift
import Foundation

enum StreamError: LocalizedError {
    case notFound(path: String)
    case accessDenied(path: String)
    case ioError(message: String)
    case storageError(reason: String)
    case streamClosed(streamId: String)
    case streamInactive(streamId: String)
    case invalidStream(streamId: String)
    case bufferError(message: String)
    
    var errorDescription: String? {
        switch self {
        case .notFound(let path):
            return "ENOENT: File does not exist: \(path)"
        case .accessDenied(let path):
            return "EACCES: Permission denied: \(path)"
        case .ioError(let message):
            return "I/O error: \(message)"
        case .storageError(let reason):
            return "Storage error: \(reason)"
        case .streamClosed(let streamId):
            return "EPIPE: Stream is closed: \(streamId)"
        case .streamInactive(let streamId):
            return "EPIPE: Stream is not active: \(streamId)"
        case .invalidStream(let streamId):
            return "ENOENT: No such stream: \(streamId)"
        case .bufferError(let message):
            return "Buffer error: \(message)"
        }
    }
} 